modin/utils.py

```diff
@@ -580,7 +579,7 @@ def import_optional_dependency(name, message):
         ) from None
 
 
-def _get_modin_deps_info() -> dict[str, JSONSerializable]:
+def _get_modin_deps_info() -> "dict[str, JSONSerializable]":
```

Comment:
Is this done to emulate the behavior of the compiler which starts storing annotations in strings starting from python 3.7?

There are also other annotation-related changes here. If these are necessary changes to support compatibility of annotations between python 3.6 and 3.7+, can you briefly describe this solution at the beginning of the pull request, so that we can see what and how to do with them, to simplify future maintaining (like BKMs).

Reply:
Should the import of `JsonSerializable` be removed from `from pandas._typing import JSONSerializable` line (line number: 32) in that case?